pkg/scheme/scheme.go

```diff
@@ -1,32 +0,0 @@
-// Copyright 2023 The Kubernetes Authors.
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//      http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-// Package scheme provides a shared scheme for Kubernetes types
-package scheme
-
-import (
-	"k8s.io/apimachinery/pkg/runtime"
-	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
-	metricsv1beta1 "k8s.io/metrics/pkg/apis/metrics/v1beta1"
-	quotamangementv1alpha1 "sigs.k8s.io/usage-metrics-collector/pkg/api/quotamanagementv1alpha1"
-)
-
-// Scheme is the scheme used by the collectors.
-var Scheme = runtime.NewScheme()
-
-func init() {
-	_ = clientgoscheme.AddToScheme(Scheme)
-	_ = metricsv1beta1.AddToScheme(Scheme)
-	_ = quotamangementv1alpha1.AddToScheme(Scheme)
-}
```

Comment:
Before removing this package and this global Scheme, the tests were failing after the update with:
```
{"level":"error","ts":"2024-05-03T22:29:00Z","logger":"usage-metrics-collector.collector","msg":"failed to list objects","error":"no kind is registered for the type quotamanagementv1alpha1.ResourceQuotaDescriptorList in scheme \"pkg/runtime/scheme.go:100\"","stacktrace":"sigs.k8s.io/usage-metrics-collector/pkg/collector.(*Collector).collect\n\t/home/prow/go/src/sigs.k8s.io/usage-metrics-collector/pkg/collector/collector.go:305\nsigs.k8s.io/usage-metrics-collector/pkg/collector.(*Collector).Collect\n\t/home/prow/go/src/sigs.k8s.io/usage-metrics-collector/pkg/collector/collector.go:196\ngithub.com/prometheus/client_golang/prometheus.(*Registry).Gather.func1\n\t/home/prow/go/pkg/mod/github.com/prometheus/client_golang@v1.16.0/prometheus/registry.go:457"}
```

I tried to trace where the Scheme was getting set and populated and couldn't fully figure out why this specific collector was still failing; after opting to just register the types with client-go's Scheme the tests started passing again, it's getting late though at this point to dig into it more 😄 I can follow-up next week in case we want to keep this package; don't have strong opinions, either way works for me